src/settings.js

```diff
@@ -0,0 +1,567 @@
+                .map(
+                    // receives the GPII setting type
+                    gpii.app.settings.settingsVisualizer.getWidgetGrade.bind(null, settingsVisualizer)
+                );
+
+            widgetGrades.forEach(function (widgetGrade) {
+                var template = fluid.defaults(widgetGrade).template;
+                resources[widgetGrade] = template;
+            });
+        }
+
+        // TODO remove this ugliness
+        var resources = {
+            rowTemplate:  fluid.defaults("gpii.app.settings.settingRow").template
+        };
+
+        appendWidgetResources(resources, settings, settingsVisualizer);
+
+        return resources;
+    };
+
+
+    /**
+     * Responsible for visualizing the list of user settings.
+     * Behaviour:
+     *  - Loads templates for the setting row and widget that is to be used
+     *  - Once templates are loaded, dynamicaly creates and supplies container for a `settingRow` element, for every user setting
+     * Expects: list of settings
+     */
+    fluid.defaults("gpii.app.settings.settingsVisualizer", {
+        gradeNames: "fluid.viewComponent",
+        model: {
+            // TODO modelRelay - transform received settings with 
+            //  pcp specific data (widgets)
+            settings: []
+        },
+        widgets: {
+            // Represents a map for GPII settings schema type to PCP widget grade to be used
+            typesToGrades: {
+                array: "multipicker",
+                boolean: "switch",
+                string: "dropDown",
+                number: "slider",
+                numberStep: "stepper",
+                text: "textfield",
+                radio: "radioToggle"
+            },
+            gradePrefix: "gpii.app.settings.widgets"
+        },
+        markups: {
+            container: "<div class=\"%containerClass\"></div>",
+            containerClassPrefix: "flc-settingListRow-%id"
+        },
+        components: {
+            resourcesLoader: {
+                type: "fluid.resourceLoader",
+                options: {
+                    resources: "@expand:gpii.app.settings.settingsVisualizer.getRequiredResources({settingsVisualizer}, {settingsVisualizer}.model.settings)",
+                    listeners: {
+                        onResourcesLoaded: {
+                            func: "{settingsVisualizer}.events.onTemplatesLoaded.fire"
+                        }
+                    }
+                }
+            },
+            // TODO
+            // Represents the list of the settings component
+            settingsLoader: {
+                type: "fluid.component",
+                createOnEvent: "onTemplatesLoaded",
+                //TODO Is this the best approach - component wrapper only to postpone initialization of dynamic components?
+                options: {
+                    dynamicComponents: {
+                        settingVisulizer: {
```

Comment:
Factor this material out into its own grade rather than a lengthy inline definition